modules/llm-cache/ds/kv_state_cache_block.cc

```diff
@@ -151,19 +151,22 @@ Status KVStateCacheBlockBuilder::Make(
 }
 
 Status KVStateCacheBlockBuilder::Query(
-    int index, std::map<int, std::pair<LLMKV, LLMKV>>& kvState) {
+    int index, std::vector<std::pair<LLMKV, LLMKV>>& kvState) {
   RETURN_ON_ASSERT((index >= 0 && index < this->blockSize),
                    "Index out of range: " + std::to_string(index));
+  RETURN_ON_ASSERT(static_cast<int>(kvState.size()) == this->layer,
+                   "The size of kvState is not equal to layer");
   for (int currentLayer = 0; currentLayer < this->layer; currentLayer++) {
-    LLMKV keyState = (kvState.find(currentLayer)->second).first;
-    LLMKV valueState = (kvState.find(currentLayer)->second).second;
-    keyState.data =
-        keyStateTensorBuilderList[currentLayer]->data() + index * tensorBytes;
-    keyState.length = tensorBytes;
-    valueState.data =
-        valueStateTensorBuilderList[currentLayer]->data() + index * tensorBytes;
-    valueState.length = tensorBytes;
-    kvState.emplace(currentLayer, std::make_pair(keyState, valueState));
+    LLMKV& keyState = kvState[currentLayer].first;
+    LLMKV& valueState = kvState[currentLayer].second;
+    memcpy(
+        keyState.data,
+        keyStateTensorBuilderList[currentLayer]->data() + index * tensorBytes,
+        tensorBytes);
+    memcpy(
+        valueState.data,
+        valueStateTensorBuilderList[currentLayer]->data() + index * tensorBytes,
+        tensorBytes);
```

Comment:
Use `vineyard::memory::concurrent_memcpy`, defined in `common/memory/memcpy.h`.

Reply:
Wait, why you need to add the memcpy to the blobstorage??

Reply:
I originally wanted the query API to be as consistent as filestorage, I will update it to the original one. Thanks.